src/index.js

```diff
@@ -24,10 +24,29 @@ class LevelDatastore {
   /* :: db: levelup */
 
   constructor (path /* : string */, opts /* : ?LevelOptions */) {
-    this.db = levelup(path, Object.assign({}, opts, {
-      compression: false, // same default as go
-      valueEncoding: 'binary'
-    }))
+    let database
+
+    if (opts && opts.db) {
+      database = opts.db
+      delete opts.db
+    } else {
+      // Default to leveldown db
+      database = require('leveldown')
+    }
+
+    this.db = levelup(
+      database(path),
+      Object.assign({}, opts, {
+        compression: false, // same default as go
+        valueEncoding: 'binary'
```

Comment:
FYI `levelup` v2 no longer includes encodings, so this `valueEncoding` option will have no effect.

Instead, wrap `database(path)` with `encoding-down`. See [UPGRADING.md](https://github.com/level/levelup/blob/HEAD/UPGRADING.md#upgrade-guide-1) for more.